File: python/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py

def potega_it(a, n):
    wynik = 1
    for i in range(n):
        wynik = wynik * a
        #print(wynik)
    
    return wynik
    

def main(args):
   # potega_it(3, 7)
    assert(potega_it(0, 2) == 0)
    assert(potega_it(2, 2) == 4)
   
    #a = int(input("podaj podstawę "))
    #n = int(input("podaj wykadnik:"))
    #potega_it(a, n)
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 * 
 * 
 * 
 */


#include <iostream>
using namespace std;


int potega_it(int a, int n) {
    int wynik = 1;
    for (int i = 1; i <= n; i++) {
        wynik = wynik *a;

        }
        return wynik;
    }

int potega_re(int a, int n) {
    if (n == 0) return 1;
    return potega_re(a, n-1) * a;
    }

int main(int argc, char **argv)
{
    int a, n;
    cout << "Podaj podstawei wykladnik: ";
    cin >> a >> n;
     potega_it(a, n); 
      potega_re(a, n); 
    cout << "potega: " << potega_it(a, n) << endl;
    cout << "potega: " << potega_re(a, n) << endl;
	
	return 0;
}

